src/ray/gcs/gcs_server/gcs_placement_group_scheduler.h

```diff
@@ -131,17 +131,68 @@ class GcsStrictSpreadStrategy : public GcsScheduleStrategy {
                        const std::unique_ptr<ScheduleContext> &context) override;
 };
 
+enum class LeasingState {
+  // TODO(sang): Use prepare and commit instead for 2PC.
+  /// The phase where lease requests haven't been returned.
+  SCHEDULING,
+  /// The phase where lease requests have returned
+  ALL_RETURNED,
+  /// Placement group has been removed, and this leasing is not valid.
+  CANCELLED
+};
+
 /// A data structure that encapsulates information regarding bundle resource leasing
 /// status.
 class LeasingContext {
-  // TODO(sang): Implement in the next PR.
+ public:
+  LeasingContext(std::shared_ptr<GcsPlacementGroup> placement_group,
+                 std::vector<std::shared_ptr<BundleSpecification>> &unplaced_bundles);
+  ~LeasingContext() = default;
+
+  bool MarkLeaseStarted(const ClientID &node_id,
+                        std::shared_ptr<BundleSpecification> bundle);
+  void MarkLeaseReturned(const ClientID &node_id,
+                         std::shared_ptr<BundleSpecification> bundle,
+                         const Status &status);
```

Comment:
Please document the public methods.

Reply:
Would you mind me doing this in the next PR? Most of signature will be changed in the atomic creation PR (I wrote this in the PR description). I can still do it if you'd want me to, so just lmk.